logger.go

```diff
@@ -72,87 +84,139 @@ const (
 	sFatal
 )
 
-func output(s severity, txt string) {
-	logLock.Lock()
-	defer logLock.Unlock()
+type logger struct {
+	infoLog  *log.Logger
+	errorLog *log.Logger
+	fatalLog *log.Logger
+	logLock  sync.Mutex
+}
+
+func (l *logger) output(s severity, txt string) {
+	l.logLock.Lock()
+	defer l.logLock.Unlock()
 	initText := "ERROR: Logging before logger.Init."
 	switch s {
 	case sInfo:
 		if !initialized {
```

Comment:
Would there be a way of implementing this using a more sensible default logger rather than accessing initialized? (Which would imply grabbing the mutex as per the above.)